### DISCUSSION TITLE
Algorithm contours returns 500: "Source data must be 1 band"

### DISCUSSION THREAD

**Discussion by davincent-ucl** [developmentseed/titiler#822] [Q&A] [answered]
Hello, 
I am quite new with TiTiler and I am trying to display the contour for a one band tiff (see attachment, in the zip file). I run the latest docker image (0.18.3) and access it locally via this url:

`http://localhost:8002/cog/tiles/WorldMercatorWGS84Quad/8/129/86?scale=1&format=tif&url=%2Fapp%2Fstorage%2F1%2Ftmp%2Fbathy.tiff&bidx=1&expression=b1&nodata=-10&unscale=true&resampling=nearest&reproject=nearest&rescale=-10%2C228&colormap_name=turbid_r&return_mask=true`

If I use the algorithm "contours" with these algorithm parameters: {"increment":10,"thickness":2,"minz":-10,"maxz":300}, the url is 

`http://localhost:8002/cog/tiles/WorldMercatorWGS84Quad/8/129/86?scale=1&format=tif&url=%2Fapp%2Fstorage%2F1%2Ftmp%2Fbathy.tiff&bidx=1&expression=b1&nodata=-10&unscale=true&resampling=nearest&reproject=nearest&algorithm=contours&algorithm_params=%7B%22increment%22%3A10%2C%22thickness%22%3A2%2C%22minz%22%3A-10%2C%22maxz%22%3A300%7D&rescale=-10%2C300&colormap_name=turbid_r&return_mask=true`

and I got this error:
"detail": "Source data must be 1 band"
The tiff has only one scalar band ranging from -10 to 228. Any idea of what i am doing wrong?
Many thanks for your help, 
David
[bathy.zip](https://github.com/user-attachments/files/15902428/bathy.zip)

**Comment by vincentsarago** [comment, marked as answer]
it's because you're trying to apply a colormap on top of the contour, which is not possible.

The `contour` algorithm will create a RGB with the `Terrain` colormap already applied

https://github.com/developmentseed/titiler/blob/77816bf93fa91f47e002aa9959ea940490224d16/src/titiler/core/titiler/core/algorithm/dem.py#L86-L90

We could totally make this colormap an algorithm option

**Reply by davincent-ucl** [reply]
Ok many thanks for your answer

**Reply by davincent-ucl** [reply]
@vincentsarago  I have another issue regarding titiler:
I am using the same tiff whose value range from -10 to 228
when I set the colormap to viridis using colormap_name, eveything works fine 

`/cog/tiles/8/129/86?rescale=-10,10&bidx=1&colormap_name=viridis&nodata=-9999&url=/app/storage/1/tmp/bathy_user_file.tiff`

But if I give my own colormap, 

`/cog/tiles/%7Bz%7D/%7Bx%7D/%7By%7D?rescale=-10,10&bidx=1&colormap=[[[-10,-6.66],[68,1,84,255]],[[-6.66,-3.33],[64,67,135,255]],[[-3.33,0],[41,120,142,255]],[[0,3.33],[34,167,132,255]],[[3.33,6.66],[121,209,81,255]],[[6.66,10],[253,231,36,255]]]&nodata=-9999&url=/app/storage/1/tmp/bathy_user_file.tiff`

the tiles displayed acts as if the negative values within the tiff are between 0 and 0,0.000000001 (i.e. in this example, the color is [34,167,132,255] rather than[68,1,84,255])
Is this an issue in titiler or am I doing something wrong?

**Reply by vincentsarago** [reply]
it's because you are using `rescale=-10,10` which then will result in values >10 to be set at 255 

if you are using you're own interval colormap, you don't need to rescale the values
